src/TraceEvent/NativeDlls.cs

```diff
@@ -15,7 +15,9 @@ class NativeDlls
     /// <param name="simpleName"></param>
     public static void LoadNative(string simpleName)
     {
-        var thisDllDir = Path.GetDirectoryName(Assembly.GetExecutingAssembly().ManifestModule.FullyQualifiedName);
+        // When TraceEvent is loaded as embedded assembly the executing assembly is mscorlib which points to the wrong path.
```

Comment:
I would like to understand this scenario more.   GetExeuctingAssembly gets the assembly of the CALLER of that API (otherwise it would be useless, as you point out it would always return mscorlib).   

Thus the existing code will look for the DLL 'next to' the assembly that compiled NativeDlls.cs (which would be Microsoft.Diagnostics.Tracing.TraceEvent.dll).  That is what I want.   There are scenarios where you don't control
the entry assembly (that is the EXE) so you don't really want to use that.   

Basically I am asking for more details about the issue that drove this change.   I think the code as it was should work.